### PR TITLE
Use Morula for setup

### PR DESCRIPTION
**Pull request by kevgo**
Hard-coding all the project names in several different places creates significant risk around forgetting to update one of those places when we add/remove/change a subproject. Better to determine the sub projects dynamically at runtime.

@trushton @hugobho

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,9 +27,6 @@ before_install:
 
 
 install:
-  - 'cd ./exosphere-shared'
```
Comment (kevgo) [inline .travis.yml:25]: Morula runs `exosphere-shared` first already.

File: cli/bin/setup [hunk 2]
```diff
@@ -7,15 +7,4 @@ if [ $? -gt 0 ]; then
...
-(cd ../exo-deploy && bin/setup && echo 'exo-deploy √')
-(cd ../exo-lint && bin/setup && echo 'exo-lint √')
-(cd ../exo-run && bin/setup && echo 'exo-run √')
-(cd ../exo-setup && bin/setup && echo 'exo-setup √')
-(cd ../exo-sync && bin/setup && echo 'exo-sync √')
-(cd ../exo-test && bin/setup && echo 'exo-test √')
-(cd ../exosphere-shared && bin/setup && echo 'exosphere-shared √')
-(node_modules/.bin/build)
-(./bin/build)
+node_modules/.bin/build
```
Comment (kevgo) [inline cli/bin/setup:10]: This repo now only prepares itself, and expects the other repos to be set up as well. The idea here is that if one changes code in this repo, it should only recompile this repo. I'm not 100% sure on this, open for feedback.

File: .travis.yml (hunk 1 above)
Comment (hugoesthere) [reply .travis.yml:25]: Where is that defined? I don't think I see it in `morula.yml`

File: cli/bin/setup (hunk 2 above)
Comment (hugoesthere) [reply cli/bin/setup:10]: I think the issue before was that the cli repo uses the rest of the `exo` commands. So when a PR is made with only changes to cli, Morula will only run tests (and `setup`) for cli. That means `setup` will not be run for the rest of the `exo` commands and their `dist` folders will not be available to the cli feature tests.

Comment (trushton) [reply cli/bin/setup:10]: Exactly what @hugobho said. On Travis, if we only made changes to the `cli` repo, we still need to setup all of the other repos too (to get their `dist` folders available) as they are required by `cli` and needed to run tests like the `tutorial.feature`.

File: .travis.yml (hunk 1 above)
Comment (kevgo) [reply .travis.yml:25]: Excellent catch! Added now.

File: cli/bin/setup (hunk 2 above)
Comment (kevgo) [reply cli/bin/setup:10]: Good catch! It now always sets up all subrepos (in https://github.com/Originate/exosphere/blob/kg-morula/.travis.yml#L30), but only tests the changed ones.

**Pull request closed without merging**